adminSiteServer/apiRoutes/dataInsights.ts

```diff
@@ -0,0 +1,294 @@
+}
+
+async function getAllDataInsightIndexItemsOrderedByUpdatedAt(
+    knex: db.KnexReadonlyTransaction
+): Promise<OwidGdocDataInsightIndexItem[]> {
+    const rows = await db.knexRaw<DataInsightRow>(
+        knex,
+        `-- sql
+
+        -- only consider published stand-alone charts since we join by slug which is only unique for published charts
+        WITH published_charts AS (
+            SELECT cc.id, cc.slug, cc.full
+            FROM chart_configs cc
+            JOIN charts c ON c.configId = cc.id
+            WHERE cc.full ->> '$.isPublished' = 'true'
+        )
+
+        SELECT
+            -- gdoc fields
+            pg.id AS gdocId,
+            pg.slug,
+            pg.content,
+            pg.published,
+            pg.publishedAt,
+            pg.markdown,
+
+            -- narrative chart fields
+            cw.id AS narrativeChartId,
+            cw.chartConfigId AS narrativeChartConfigId,
+
+            -- chart config (prefer narrative charts over grapher URLs)
+            COALESCE(cc_chartView.full, cc_grapherUrl.full) AS chartConfig,
+
+            -- image fields
+            i.id AS imageId,
+            i.filename,
+            i.cloudflareId,
+            i.originalWidth
+        FROM posts_gdocs pg
+
+        -- extract the slug from the given Grapher URL and join by it
+        LEFT JOIN published_charts cc_grapherUrl
+            ON cc_grapherUrl.slug = SUBSTRING_INDEX(SUBSTRING_INDEX(content ->> '$."grapher-url"', '/grapher/', -1), '\\?', 1)
+
+        -- join the chart_views table to get the config of the narrative chart
+        LEFT JOIN chart_views cw
+            ON cw.name = content ->> '$."narrative-chart"'
+        LEFT JOIN chart_configs cc_chartView
+            ON cc_chartView.id = cw.chartConfigId
+
+        -- join the images table by filename (only works for data insights where the image block comes first)
+        LEFT JOIN images i
+            ON i.filename = COALESCE(pg.content ->> '$.body[0].smallFilename', pg.content ->> '$.body[0].filename')
+
+        WHERE
+            pg.type = 'data-insight'
+            AND i.replacedBy IS NULL
+
+        ORDER BY pg.updatedAt DESC`
+    )
+    const groupedTags = await getTagsGroupedByGdocId(
+        knex,
+        rows.map((row) => row.gdocId)
+    )
+    return rows.map((row) =>
+        extractDataInsightIndexItem({
+            ...row,
+            tags: groupedTags[row.gdocId] ? groupedTags[row.gdocId] : undefined,
+        })
+    )
+}
+
+function extractDataInsightIndexItem(
+    dataInsight: DataInsightRow
+): OwidGdocDataInsightIndexItem {
+    const content = parsePostGdocContent(
+        dataInsight.content
+    ) as OwidGdocDataInsightContent
+    const chartConfig = dataInsight.chartConfig
+        ? parseChartConfig(dataInsight.chartConfig)
+        : undefined
+
+    // check if the given grapher-url is a valid Grapher or Explorer URL
+    const grapherUrlField = content["grapher-url"]?.trim()
```

Comment:
If trimming is necessary here, we may as well fix it upstream

https://github.com/owid/owid-grapher/blob/d132fba478a7c123973a3ecaadb5b8f8a74b1d07/db/model/Gdoc/archieToEnriched.ts#L313-L319